### PR TITLE
Rename ContextType to Target in ClientConfig

### LINKED ISSUE #4073: Rename `ContextType` to `Target` in ClientConfig API
**Describe the feature request**
With Context/Target implementation we have added a `Context-Type` field to the ClientConfig API. 

There is also another term called `Target` that we have been using to denote the same thing. Based on some of the discussions we have agreed to use `Target` term at all places. So, this issue tracks work needed to rename the `ContextType` in the ClientConfig API to target.

Considering, current `Context/Target` support feature is under feature-flag and has been disabled by default, we will be able to rename this without any backward compatibility issues.

**Describe alternative(s) you've considered**

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

We have been using 2 different terms, `ContextType` and `Target` to specify the same thing. Considering each context is associated with a `Target` instead of using `ContextType` we are renaming it to `Target`.

**Note:** This will require users to delete and recreate existing TMC context if there is any present. I would recommend to remove `config-ng.yaml` before testing this if that is the case.

**Before:**
```
contexts:
    - name: tmc-unstable
      type: tmc
      ...
    - name: kind-test-cluster
      type: k8s
      ...
currentContext:
    tmc: tmc-unstable
    k8s: kind-test-cluster
```

**After:**
```
contexts:
    - name: tmc-unstable
      target: mission-control    <-------
      ...
    - name: kind-test-cluster
      target: kubernetes         <-------
      ...
currentContext:
    mission-control: tmc-unstable     <-------
    kubernetes: kind-test-cluster     <-------
```

### Which issue(s) this PR fixes

Fixes #4073

### Describe testing done for PR

- `tanzu config get` and verified the output
```
$ tanzu config get | grep -A2 -B2 target
contexts:
    - name: tmc-unstable
      target: mission-control
      globalOpts:
        endpoint: unstable.tmc-dev.cloud.vmware.com:443
--
            type: api-token
    - name: kind-test-cluster
      target: kubernetes
      clusterOpts:
        path: /Users/anujc/.kube/config
--
        isManagementCluster: true
    - name: kind-test-cluster-2
      target: kubernetes
      clusterOpts:
        path: /Users/anujc/.kube/config

```
- List the available context with different flags
```
$ tanzu context list
Target:  kubernetes
  NAME                 ISACTIVE  ENDPOINT  KUBECONFIGPATH             KUBECONTEXT        
  kind-test-cluster    false               /Users/anujc/.kube/config  kind-test-cluster  
  kind-test-cluster-2  true                /Users/anujc/.kube/config  kind-test-cluster  
Target:  mission-control
  NAME          ISACTIVE  ENDPOINT                               
  tmc-unstable  false     unstable.tmc-dev.cloud.vmware.com:443  
```
```
$ tanzu context list --target tmc
Target:  mission-control
  NAME          ISACTIVE  ENDPOINT                               
  tmc-unstable  false     unstable.tmc-dev.cloud.vmware.com:443 
```
```
$ tanzu context list --target mission-control
Target:  mission-control
  NAME          ISACTIVE  ENDPOINT                               
  tmc-unstable  false     unstable.tmc-dev.cloud.vmware.com:443  
```
```
$ tanzu context list --target k8s
Target:  kubernetes
  NAME                 ISACTIVE  ENDPOINT  KUBECONFIGPATH             KUBECONTEXT        
  kind-test-cluster    false               /Users/anujc/.kube/config  kind-test-cluster  
  kind-test-cluster-2  true                /Users/anujc/.kube/config  kind-test-cluster  
```
```
$ tanzu context list --target random
✖  invalid target specified. Please specify correct value of `--target` or `-t` flag from 'kubernetes/k8s/mission-control/tmc' 
```

### Release note

```release-note
Rename ContextType to Target in ClientConfig API
```

### CODE REVIEW

File: cli/runtime/config/nodeutils/create_node.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-// CreateNode creates yaml node based on node key Type
+// CreateNode creates yaml node based on node key Target
```
Comment (mpanchajanya) [inline cli/runtime/config/nodeutils/create_node.go:10]: nit; Type here is referring to yaml node so we can keep Type as is

**Review by mpanchajanya — COMMENTED** [review]
Looks good to me. Thanks Anuj

**Review by mpanchajanya — APPROVED** [review]

File: cli/core/pkg/command/context.go [hunk 2]
```diff
@@ -64,7 +65,7 @@ func init() {
 
 	initCreateCtxCmd()
 
-	listCtxCmd.Flags().StringVarP(&ctxType, "type", "t", "", "context type (k8s|tmc)")
```
Comment (anujc25) [inline cli/core/pkg/command/context.go:67]: Need to support `k8s/tmc` inputs. Looks like it doesn't support it right now. Also need a validation if user enters invalid target, it should throw and error. Currently, it returns empty list.

```
$ tanzu context list -t k8s
```
```
$ tanzu context list -t kubernetes
Target:  kubernetes
  NAME                 ISACTIVE  ENDPOINT  KUBECONFIGPATH             KUBECONTEXT        
  kind-test-cluster    false               /Users/anujc/.kube/config  kind-test-cluster  
  kind-test-cluster-2  true                /Users/anujc/.kube/config  kind-test-cluster  
  ```
  ```
  $ tanzu context list -t random
```

File: cli/core/pkg/command/context.go [hunk 3]
```diff
@@ -64,7 +65,7 @@ func init() {
...
+	listCtxCmd.Flags().StringVarP(&target, "target", "t", "", "list only contexts associated with the specified target (kubernetes[k8s]|mission-control[tmc])")
```
Comment (anujc25) [inline cli/core/pkg/command/context.go:68]: Need to support `k8s/tmc` inputs. Looks like it doesn't support it right now. Also need a validation if user enters invalid target, it should throw and error. Currently, it returns empty list.

```
$ tanzu context list -t k8s
```
```
$ tanzu context list -t kubernetes
Target:  kubernetes
  NAME                 ISACTIVE  ENDPOINT  KUBECONFIGPATH             KUBECONTEXT        
  kind-test-cluster    false               /Users/anujc/.kube/config  kind-test-cluster  
  kind-test-cluster-2  true                /Users/anujc/.kube/config  kind-test-cluster  
  ```
  ```
  $ tanzu context list -t random
```

**Review by marckhouzam — APPROVED** [review]
Nicely done!
One single nit.

LGTM

File: cli/core/pkg/command/context.go [hunk 4]
```diff
@@ -64,7 +65,7 @@ func init() {
 
 	initCreateCtxCmd()
 
-	listCtxCmd.Flags().StringVarP(&ctxType, "type", "t", "", "context type (k8s|tmc)")
+	listCtxCmd.Flags().StringVarP(&targetStr, "target", "t", "", "list only contexts associated with the specified targetStr (kubernetes[k8s]|mission-control[tmc])")
```
Comment (marckhouzam) [inline cli/core/pkg/command/context.go:68]:
```suggestion
	listCtxCmd.Flags().StringVarP(&targetStr, "target", "t", "", "list only contexts associated with the specified target (kubernetes[k8s]|mission-control[tmc])")
```

File: cli/core/pkg/command/context.go [hunk 5]
```diff
@@ -681,9 +682,13 @@ func displayContextListOutputSplitViewTarget(cfg *configapi.ClientConfig, writer
 
 	cyanBold := color.New(color.FgCyan).Add(color.Bold)
 	cyanBoldItalic := color.New(color.FgCyan).Add(color.Bold, color.Italic)
-	_, _ = cyanBold.Println("Target: ", cyanBoldItalic.Sprintf("%s", configapi.CtxTypeK8s))
-	outputWriterK8sTarget.Render()
+	if target == "" || target == string(cliv1alpha1.TargetK8s) {
```
[the remark this answers is not in the document]
Comment (anujc25) [reply cli/core/pkg/command/context.go]: Yeah. I didn't even know there was a flag to filter context by target (`type` earlier) with `tanzu context list`.

[the remark this answers is not in the document]
Comment (marckhouzam) [reply cli/core/pkg/command/context.go]: Me neither 🤯

**Review by vuil — APPROVED** [review]
LGTM, thanks!

**Pull request merged**